src/BenchmarkDotNet.Toolchains.Uap/Templates/UapBenchmarkProgram.txt

```diff
@@ -0,0 +1,285 @@
+$TargetMethodReturnTypeNamespace$
+
+$AdditionalLogic$
+
+namespace BenchmarkDotNet.Autogenerated
+{
+    class LoggingChannelTextWriterAdapter : TextWriter
+    {
+        private readonly LoggingChannel lc;
+
+        public LoggingChannelTextWriterAdapter(LoggingChannel lc)
+        {
+            this.lc = lc;
+        }
+
+        public override Encoding Encoding => Encoding.UTF8;
+
+        public override void Write(char value)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override void Write(char[] buffer, int index, int count)
+        {
+            lc.LogMessage(new string(buffer, index, count));
+        }
+    }
+
+    public static class Program
+    {
+        static void Main(string[] args)
+        {
+            global::Windows.UI.Xaml.Application.Start((p) => new App(args));
+        }
+        }
+        
+        $ExtraAttribute$
+    sealed partial class App : Application
+    {
+        LoggingChannel lc = new LoggingChannel("BenchmarkDotNet.Autogenerated", null, new Guid("4bd2826e-54a1-4ba9-bf63-92b73ea1ac4a"));
+
+        public App(string[] args)
+        {
+            this.InitializeComponent();
+
+            bool isDiagnoserAttached = args != null && args.Contains(BenchmarkDotNet.Engines.Engine.Signals.DiagnoserIsAttachedParam);
+            ConsoleHost host = new ConsoleHost(new LoggingChannelTextWriterAdapter(lc), isDiagnoserAttached);
+
+            // the first thing to do is to let diagnosers hook in before anything happens
+            // so all jit-related diagnosers can catch first jit compilation!
+            if (isDiagnoserAttached)
+                host.BeforeAnythingElse();
+
+            try
+            {
+                Action action = () =>
+                {
+                    try
+                    {
+                        // we are not using Runnable here in any direct way in order to avoid strong dependency Main<=>Runnable
+                        // which could cause the jitting/assembly loading to happen before we do anything
+                        // we have some jitting diagnosers and we want them to catch all the informations!!
+
+                        Type type = typeof(Program).GetTypeInfo().Assembly.GetType("BenchmarkDotNet.Autogenerated.Runnable");
+
+                        type.GetMethod("Run", BindingFlags.Public | BindingFlags.Static).Invoke(null, new object[] { host });
+                    }
+                    catch (Exception ex)
+                    {
+                        lc.LogMessage(string.Format("// Exception #1: {0}. StackTrace: {1}", ex.Message, ex.StackTrace));
+                        throw;
+                    }
+                    finally
+                    {
+                        lc.Dispose();
+                        Application.Current.Exit();
+                    }
+                };
+
+                Task.Factory.StartNew(action);
+            }
+            catch (Exception ex)
```

Comment:
the second catch will never be raised. You start the async job and as long as you don't await/wait the task it will not throw in this context. Suggestion: just remove the second catch block